From 8c40930e1be81c709fb2ab2369f5d61618bd37ff Mon Sep 17 00:00:00 2001
From: Redbeanw44602 <redbeana44945@gmail.com>
Date: Wed, 19 Aug 2026 15:32:51 +0800
Subject: [PATCH] fix.

---
 .../UserMode/PlatformUtil/Linux/ProcessRuntimeUtility.cc   | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntimeUtility.cc b/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntimeUtility.cc
index 862c29f..22a1887 100644
--- a/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntimeUtility.cc
+++ b/source/Backend/UserMode/PlatformUtil/Linux/ProcessRuntimeUtility.cc
@@ -153,15 +153,16 @@ static tinystl::vector<RuntimeModule> &get_process_map_with_proc_maps() {
     if (strcmp(permissions, "r--p") != 0 && strcmp(permissions, "r-xp") != 0)
       continue;
 
+    char *path_buffer = line_buffer + path_index;
+    if (*path_buffer == 0 || *path_buffer == '\n' || *path_buffer == '[')
+      continue;
+
     // check elf magic number
     ElfW(Ehdr) *header = (ElfW(Ehdr) *)region_start;
     if (memcmp(header->e_ident, ELFMAG, SELFMAG) != 0) {
       continue;
     }
 
-    char *path_buffer = line_buffer + path_index;
-    if (*path_buffer == 0 || *path_buffer == '\n' || *path_buffer == '[')
-      continue;
     RuntimeModule module;
 
     // strip
-- 
2.55.0

